Geometry_Engine/Query/HashArray.cs

```diff
@@ -66,100 +86,173 @@ public static double[] IHashArray(this IGeometry igeometry)
 
+            if (!subParts.Any())
+                if (curve is Polyline)
+                    return HashArray(curve.IControlPoints(),
+                        translationFactor,
+                        comparisonConfig: comparisonConfig,
+                        fullName: fullName.AppendPropertyName($"{nameof(ControlPoints)}"));
+                else
+                    return HashArray(curve as dynamic,
+                        translationFactor,
+                        skipEndPoint: false,
+                        comparisonConfig: comparisonConfig,
+                        fullName: fullName);
+
             List<double> hashes = new List<double>();
 
             //Add hash ignoring endpoint for all but last curve
             for (int i = 0; i < subParts.Count - 1; i++)
             {
-                hashes.AddRange(HashArray(subParts[i] as dynamic, translationFactor, true));
+                hashes.AddRange(HashArray(subParts[i] as dynamic,
+                    translationFactor,
+                    skipEndPoint: true,
+                    comparisonConfig: comparisonConfig,
+                    fullName: fullName.AppendPropertyName($"[{i}]"))
+                );
             }
-            //Include endpoint for hasing for last curve
-            hashes.AddRange(HashArray(subParts.Last() as dynamic, translationFactor, false));
+
+            //Include endpoint for hashing for last curve
+            hashes.AddRange(HashArray(subParts.Last() as dynamic,
+                translationFactor,
+                skipEndPoint: false,
+                comparisonConfig: comparisonConfig,
+                fullName: fullName.AppendPropertyName($"[{subParts.Count - 1}]")
+                )
+            );
 
             return hashes.ToArray();
         }
 
         /***************************************************/
 
         [Description("The GeometryHash for an Arc is calculated as the GeometryHash of the start, end and middle point of the Arc.")]
-        private static double[] HashArray(this Arc curve, double translationFactor, bool skipEndPoint = false)
+        private static double[] HashArray(this Arc curve, double translationFactor, bool skipEndPoint = false, BaseComparisonConfig comparisonConfig = null, string fullName = null)
         {
+            if (comparisonConfig?.TypeExceptions?.Any(t => typeof(ICurve).IsAssignableFrom(t)) ?? false)
+                return default;
+
             translationFactor += (int)TypeTranslationFactor.Arc;
 
-            IEnumerable<double> hash = curve.StartPoint().HashArray(translationFactor)
-               .Concat(curve.PointAtParameter(0.5).HashArray(translationFactor));
+            IEnumerable<double> hash = curve.StartPoint().HashArray(translationFactor,
+                    comparisonConfig,
+                    fullName.AppendPropertyName($"{fullName}.{nameof(StartPoint)}"))
+               .Concat(curve.PointAtParameter(0.5).HashArray(translationFactor,
```

Comment:
Is that used to enable avoiding checking curves for particular control points? If so, I would argue it is impractical, would be happy to skip this level of complexity in favour of code readability, assuming that no one will ever compare arcs ignoring their mid point (especially that the consumer would somehow need to know that arcs get sampled at parameter equal to 0.5)

Reply:
The middle point is required to compare two arcs. The GeometryHash for an Arc is calculated as the GeometryHash of the start, end and middle point of the Arc.

Reply:
Ah sorry I didn't understand! Yes you're right then -- it's may be a bit overzealous, I just went straight for the most flexible mechanism.